lib/index.d.ts

```diff
@@ -0,0 +1,228 @@
+/// <reference types="node" />
+
+export interface RenderOptions {
+  page?: number
+  density?: number
+  quality?: number | undefined
+  outline?: boolean
+}
+
+export interface SaveOptions extends RenderOptions {
+  format?: string | undefined
+  matte?: string
+}
+
+export class Canvas {
+  /** @internal */
+  static contexts: WeakMap<Canvas, readonly CanvasRenderingContext2D[]>
+
+  constructor(width?: number, height?: number)
+
+  /**
+   * Cast this object as a {@link SyncCanvas} and set the `async` property to false to get the synchronous behaviours.
+   *
+   * @example
+   * import { Canvas, SyncCanvas } from "."
+   * const myCanvas = new Canvas() as any as SyncCanvas
+   * myCanvas.async = false
+   * const result = myCanvas.toBuffer("png") // now these functions return synchronously
+   */
+  async: true
+
+  get width(): number
+  get height(): number
+
+  getContext(type?: "2d"): CanvasRenderingContext2D
+
+  get pages(): readonly CanvasRenderingContext2D[]
+
+  get pdf(): Promise<Buffer>
+  get svg(): Promise<Buffer>
+  get jpg(): Promise<Buffer>
+  get png(): Promise<Buffer>
+
+  newPage(width?: number, height?: number): CanvasRenderingContext2D
+
+  saveAs(filename: string, options?: SaveOptions): Promise<void>
```

Comment:
One issue with encapsulating the various options into interfaces into types is that (at least in my testing) VS Code gives much less helpful prompts in its tooltips. For instance here's what I get with your current version: 
<img width="726" alt="image" src="https://user-images.githubusercontent.com/469523/144895520-498fa3e1-9831-467c-930d-a970feff256d.png">

But if they're fully enumerated it displays: 
<img width="830" alt="image" src="https://user-images.githubusercontent.com/469523/144895805-d88a0f69-0ace-43e7-97e0-1b5ed327bcf8.png">

Reply:
Yeah, I usually do it like that because -from what I've seen in DefinitelyTyped-, it's idiomatic. There's no rule written anywhere for either of them, so we can change it if it would fit the library better.

(Usually you would reach until opening the curly brace and then press `Ctrl+Space` so you'd get autocomplete prompts. To look up `SaveOptions` fully, I usually `Cmd+Click` on the function, and then again on the type you're interested to see its full definition.)

Reply:
Ah great, `ctrl+space` is exactly the kind of UI prompting I was hoping for with the options.